src/plugins/formulas/engine/settings.js

```diff
@@ -0,0 +1,81 @@
+import { PLUGIN_KEY } from '../formulas';
+
+// TODO: refine the list of default HF settings
+const DEFAULT_SETTINGS = {
+  licenseKey: 'internal-use-in-handsontable',
+
+  // default engine configuration
+  binarySearchThreshold: 20,
+  matrixDetection: true,
+  matrixDetectionThreshold: 100,
+  useColumnIndex: false,
+  useStats: false,
```

Comment:
`matrixDetection` to false
```suggestion
  binarySearchThreshold: 20,
  matrixDetection: false,
  matrixDetectionThreshold: 100,
  useColumnIndex: false,
  useStats: false,
```